Automattic/wp-calypso
# Move traintracks analytics out of suggestions

We hope to extract the `Suggestions` component for usage outside of Calypso.

#31466 added an analytics call in suggestions `Item` component based on presence of a `railcar` prop. This PR pushes that handling up tot he `SuggestionsSearch` component, although ideally it could be handled entirely outside of these shared components [in the consumer](https://github.com/Automattic/wp-calypso/blob/5ae9ef1b7257015ba69f21256b51c3ab9aa6acce/client/components/site-verticals-suggestion-search/index.jsx).

Pushing the analytics out of `Suggestions` is a first step and sufficient to be able to extract a generic `Suggestions` component for use outside of Calypso.

Implemented with the help of @razvanpapadopol 

#### Screens

This is the affected suggestions list:

![Screen Shot 2019-11-07 at 11 38 14](https://user-images.githubusercontent.com/841763/68382312-be9b8d00-0153-11ea-90dd-45df927e8e95.png)


#### Testing instructions

* Confirm the `Item` is only used by `Suggestions`, and can be modified freely.
* Confirm that `Suggestions` is only affected when it receives a `railcar` prop, and that this behavior is equivalent. The only usage we found was in the mentioned `SiteVerticalsSuggestionSearch` component, which goes through `SuggestionsSearch`.

Manual testing

We'll compare the debug output of tracks between production and our branch to confirm the behavior is maintained. Do the following steps on WordPress.com and on this branch.

- Visit `/start/site-type`
- set `localStorage.debug = 'calypso:analytics:tracks'` in your console. You may want to filter console output with a term like `traintracks`, the events we'll observe are `calypso_traintracks_render` and `calypso_traintracks_interact`
- refresh
- Pick a site type
- At the suggestion screen, type a search screen. You should see identical⋆ tracks output.
- Click on a suggestion, again tracks output should be identical*.

⋆ The tracks event will include a `railcar` property that has a unique prefix like  `9ba25b2f0c1c4cdb93a84aadd668a3cf-site-vertical-suggestion-5`. `-site-vertical-suggestion-5` should be identical when clicking on the same item

Confirm there are no regressions where `Suggestions` is used:
- `client/extensions/zoninator/components/search-autocomplete/post-suggestions.jsx`
- `client/my-sites/domains/domain-management/transfer/transfer-out/select-ips-tag.jsx`

Confirm there are no regressions where `SuggestionSearch` is used. The already mentioned `client/components/site-verticals-suggestion-search/index.jsx` seems to be the only place it's used.
